### PR TITLE
Honor the collect_ignore option used in conftest.py

### PR DESCRIPTION
**Pull request by oscarbenjamin**
Replacement for #34

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -1,6 +1,8 @@
 0.3.0 (unreleased)
 ==================
 
+- Honor the ``collect_ignore`` option used in ``conftest.py``. [#34]
```
Comment (drdavella) [inline CHANGES.rst]: Just a nit, but we need to change the PR number here now.

Comment (oscarbenjamin) [reply CHANGES.rst]: Okay, I've just fixed that and force-pushed.

**Comment by oscarbenjamin** [comment]
I've pushed another change because it would have failed the tests - if `collect_ignore` is not specified in `conftest.py` then `_getcontest_pathlist` returns `None` rather than an empty list.

**Comment by drdavella** [comment]
Thank you @oscarbenjamin!

**Pull request merged**